lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -51,6 +55,63 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+// clang-format off
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+
+    (inputs)
+    (varname)
+);
+// clang-format on
+
+namespace {
+// We want to know if this material is a specialization that was created to handle UV mappings on
+// export. For details, see the _UVMappingManager class in ..\shading\shadingModeExporterContext.cpp
+//
+bool _IsMergeableMaterial(const UsdShadeMaterial& shadeMaterial)
+{
+    if (!shadeMaterial || !shadeMaterial.HasBaseMaterial()) {
+        return false;
+    }
+
+    UsdPrimCompositionQuery                 query(shadeMaterial.GetPrim());
+    std::vector<UsdPrimCompositionQueryArc> arcs = query.GetCompositionArcs();
+
+    // Check for materials created by _UVMappingManager::getMaterial(). This code could probably be
+    // expanded to be more generic and handle more complex composition arcs at a later stage.
+
+    // Materials created by the _UVMappingManager have only 2 arcs:
+    if (arcs.size() != 2) {
+        return false;
+    }
+
+    UsdPrimCompositionQueryArc specializationArc = arcs[1];
+    SdfPath                    primPath = shadeMaterial.GetPath();
+
+    // Check that the specialization arc contains only opinions on varname inputs:
+    bool retVal = true;
+    auto isVarnameOpinion = [&](const SdfPath& path) {
+        if (path == primPath) {
+            return;
+        }
+        if (path.GetParentPath() != primPath || !path.IsPrimPropertyPath()) {
+            retVal = false;
+            return;
+        }
+        std::vector<std::string> splitName = SdfPath::TokenizeIdentifier(path.GetName());
+        // We allow only ["inputs", "<texture_name>", "varname"]
+        if (splitName.size() != 3 || splitName[0] != _tokens->inputs.GetString()
+            || splitName[2] != _tokens->varname.GetString()) {
+            retVal = false;
+        }
+    };
+
+    specializationArc.GetIntroducingLayer()->Traverse(primPath, isVarnameOpinion);
```

Comment:
The layer-based traversal at the bottom here has the potential to be way more expensive than it needs to be. At the point where we start the traversal, all we know is that we're looking at a specialized material. The root of the traversal is the primSpec in the layer that introduced the specialize arc. That layer may also have an arbitrary number of properties and/or namespace children authored for that Material, and we're going to end up visiting **all** of those paths when really we know whether or not we should stop after inspecting only the Material primSpec and its properties.

With the caveat that this is untested, this is kind of what I'm thinking for this, beginning right after we verify that there are only two arcs:

```C++
    // This is a little more robust than grabbing a specific arc index.
    UsdPrimCompositionQuery::Filter filter;
    filter.arcTypeFilter = UsdPrimCompositionQuery::ArcTypeFilter::Specialize;
    query.SetFilter(filter);
    arcs = query.GetCompositionArcs();
    const UsdPrimCompositionQueryArc specializationArc = arcs.front();

    const SdfLayerHandle layer = specializationArc.GetIntroducingLayer();
    const SdfPrimSpecHandle primSpec = layer->GetPrimAtPath(shadeMaterial.GetPath());

    // If the primSpec that specializes the base material introduces other
    // namespace children, it can't be merged.
    if (!primSpec->GetNameChildren().empty()) {
        return false;
    }

    // Check that the only properties authored are varname inputs.
    for (const SdfPropertySpecHandle& propSpec : primSpec->GetProperties()) {
        const SdfPath propPath = propSpec->GetPath();

        const std::vector<std::string> splitName =
            SdfPath::TokenizeIdentifier(propPath.GetName());
        // We allow only ["inputs", "<texture_name>", "varname"]
        if (splitName.size() != 3u ||
                splitName[0u] != _tokens->inputs.GetString() ||
                splitName[2u] != _tokens->varname.GetString()) {
            return false
        }
    }

    return true;
```